### DISCUSSION TITLE
Designer inside Visual Studio is not working for control in library

### DISCUSSION THREAD

**Discussion by lukaspirkl** [AvaloniaUI/Avalonia#12341] [Q&A] [answered]
I am trying to create an application with plugins. I have a library (a plugin) that is referencing Avalonia and containing only the UserControl that is displayed by the application. The application does not reference the library directly but it is loaded at run time. I found out that the designer in Visual Studio needs to have an executable to be able to provide IntelliSense and display preview. So my idea is that each plugin will have a dummy Avalonia application just to reference the library for the designer. It looks like it should be working but the designer is now displaying `System.Xaml.XamlException: Unable to resolve type HelloPlugin.UserControl1`. 
![image](https://github.com/AvaloniaUI/Avalonia/assets/779950/bfc388e2-714d-440e-a478-301c961a89c1)
There is the code: [DotNetPlugins.zip](https://github.com/AvaloniaUI/Avalonia/files/12173769/DotNetPlugins.zip)

I tried to reproduce this in the simplest possible way, but for some reason, it is working. For comparison, this is the code that is working correctly: [AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/12173792/AvaloniaApplication1.zip)

Why it is unable to resolve the type in the first case but it is working correctly in the second?

**Comment by kekekeks** [comment]
Previewer needs your control library to be referenced (and preferably loaded) by an executable project. Try adding `GC.KeepAlive(typeof(Type.From.Your.Library));` at the beginning of BuildAvaloniaApp.

**Reply by lukaspirkl** [reply]
The library is referenced. I tried to add the line you mentioned but the result is still the same.

**Comment by lukaspirkl** [comment, marked as answer]
The reason why this is not working is missing reference to another library. I was following the [Microsoft guide to creating the plugin project](https://learn.microsoft.com/en-us/dotnet/core/tutorials/creating-app-with-plugin-support#simple-plugin-with-no-dependencies). There is information that the `ProjctReference` should be added with `Private="false"` and `ExcludeAssets="all"` which is causing that the referenced project is not copied into the output. When I remove these two settings the type is found correctly and the previewer is happy.
